src/dataStructures/linkedList/README.md

```diff
@@ -0,0 +1,100 @@
+
+*Source: BeginnersBook*
+
+### Search
+
+Due to the contiguous nature of an array, we only need the memory address for the first element
+to calculate the address for any subsequent element, resulting in a *O(1)* lookup time.
+
+However, to look for a element in a linked list of size *n*, we would have to traverse through each node
+to get the pointer to the next node until we reached our required node. This creates a *O(n)* lookup time.
+
+There are also some unique methods to speed up your linked lists searches, which will be discussed in a later section.
+
+**Generally, searching a index in a array is faster than a linked list.**
+
+### Insert
+
+#### Front & End Insertion
+When we try to insert a element from the beginning and the end of array, to keep the contiguity in memory,
+we need to shift all other elements to the left (in the case of a end insertion)
+or right (in the case of a front insertion). This incurs a *O(n)* time complexity.
+
+However, we just need to update the pointer address in our new node (for a front insertion), or a 
+pointer address in our last node (for a end insertion) in a linked list, incurring a *O(1)* time complexity.
+
+#### Inserting into the *i-th* index
+To insert a element into the *i-th* position of an array, we need to navigate to that position and shift all element
+that have a index of more than *i + 1* back.
+
+For a linked list, we have to navigate to the *i - 1 th* node and change the pointer to reference our new node. 
+We can then reference the rest of the list by updating the pointer in our newly added node.
+
+In both cases, a *O(n)* time complexity is incurred.
+
+**Generally, inserting a element into a linked list is faster than an array.**
+
+### Memory Requirements & Flexibility
+As a contiguous block of memory must be allocated for an array, its size is fixed.
+If we declare a array of size *n*, but only allocate *x* elements, where *x < n*, 
+we will be left with unused, wasted memory.
+
+This waste does not occur with linked lists, which only take up memory as new elements are added.
+However, additional space will be used to store the pointers. 
+
+As the declared size of our array is static (done at compile time), we are also given less flexibility if 
+we end up needing to store more elements at run time.
+
+However, linked list gives us the option of adding new nodes at run time based on our requirements, 
+allowing us to allocate memory to store items dynamically, giving us more flexibility.
+
+**Generally, linked list is more memory efficient and flexible than arrays.**
+
+### Conclusion
+You should aim to use linked list in scenarios where you cannot predict how many elements you need to store
+or if you require constant time insertions to the list.
+
+However, arrays would be preferred if you already know the amount of elements you need to store ahead of time. 
+It would also be preferred if you are conducting a lot of look up operations.
+
+## Linked List Variants
```

Comment:
Do mention double-linked list too. This is the most common variant after the conventional linked list, that allows popping of elements from front and back. Java's Deque (double-ended queue i believe) is essentially a double-linkedlist